### PR TITLE
Adding support for application availability checks

### PR DESCRIPTION
**Pull request by abellotti**
Now triggering application availability checks with the source check_availability action.

### CODE REVIEW

**Comment by abellotti** [comment]
Marking as WIP because:

- [ ] Depends on https://github.com/RedHatInsights/sources-api/pull/128
- [ ] needs to be tested
- [ ] needs specs
- [ ] need to review post signature to the micro-services

File: app/controllers/api/v1/sources_controller.rb [hunk 1]
```diff
@@ -15,6 +18,49 @@ def create
...
+
+          headers = {
+            "Content-Type"  => "application/json",
+            "x-rh-identity" => Base64.strict_encode64({'identity' => { 'account_number' => source.tenant.external_tenant }}.to_json)
+          }
+
+          uri = URI.parse(url)
+          net_http = Net::HTTP.new(uri.host, uri.port)
+          request  = Net::HTTP::Post.new(uri.request_uri, headers)
+          request.body = { "source_id" => source.id.to_s }.to_json
```
Comment (abellotti) [inline app/controllers/api/v1/sources_controller.rb]: /cc @agrare just initial prototype here, we need to design/agree on signature/contract with the apps.

Comment (agrare) [reply app/controllers/api/v1/sources_controller.rb]: :+1: this is what we discussed with @dccurtis

**Comment by abellotti** [comment]
Ok, added specs, can you lend some 👀 @agrare ?

**Pull request merged**